app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapCoordinateFieldToFeatureCollection.kt

```diff
@@ -0,0 +1,76 @@
+package org.dhis2.uicomponents.map.geometry.mapper.featurecollection
+
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import org.dhis2.data.dhislogic.CoordinateAttributeInfo
+import org.dhis2.data.dhislogic.CoordinateDataElementInfo
+import org.dhis2.data.dhislogic.CoordinateFieldInfo
+import org.dhis2.uicomponents.map.geometry.mapper.feature.MapCoordinateFieldToFeature
+
+class MapCoordinateFieldToFeatureCollection(
+    private val mapCoordinateFieldToFeature: MapCoordinateFieldToFeature
+) {
```

Comment:
you're missing some flows in the unit test. I think if you can split this class everything will be eassier For ej: Mapattribute and dataelement could be 2 classes

Reply:
👍